third_party/move/move-core/types/src/language_storage.rs

```diff
@@ -67,6 +68,15 @@ pub enum TypeTag {
     U32,
     #[serde(rename = "u256", alias = "U256")]
     U256,
+
+    // NOTE: added in bytecode version v8
+    Function(
+        #[serde(
+            serialize_with = "safe_serialize::type_tag_recursive_serialize",
+            deserialize_with = "safe_serialize::type_tag_recursive_deserialize"
+        )]
+        Box<FunctionTag>,
```

Comment:
I really hate boxes that we use everywhere in tree data structures and then we end up cloning them over and over and over. Can we switch to Arcs here, also see discussion on slack.